python/neuroglancer/local_volume.py

```diff
@@ -244,3 +251,80 @@ def invalidate(self):
             self._mesh_generator_pending = None
             self._mesh_generator = None
         self._dispatch_changed_callbacks()
+
+class DataWrapper:
+    """Wraps data for LocalVolume."""
+
+    def __init__(self, data):
+        self._data = data
+        self._default_labels = ['d%d' % d for d in range(self.rank)]
+        self._default_origin = [0] * self.rank
+        self._default_scales = [1] * self.rank
+        self._default_units = [''] * self.rank
+
+    def __getitem__(self, item):
+        return self._data.__getitem__(item)
+
+    def __getattr__(self, attr):
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        return getattr(self._data, attr)
+
+    @property
+    def labels(self):
+        return self._default_labels
+
+    @property
+    def origin(self):
+        return self._default_origin
+
+    @property
+    def rank(self):
+        return len(self._data.shape)
+
+    @property
+    def scales(self):
+        return self._default_scales
+
+    @property
+    def units(self):
+        return self._default_units
+
+
+class TensorStoreDataWrapper(DataWrapper):
+    """Wraps TensorStore data for LocalVolume."""
+
+    def __init__(self, data):
+        super().__init__(data=data[ts.d[:].translate_to[0]])
+        self.dtype = self._data.dtype.numpy_dtype
+        self._origin_before_translation = data.origin
+
+    def __getitem__(self, item):
+        return self._data.__getitem__(item).read().result()
+
+    @property
+    def labels(self):
+        if any(self._data.domain.labels):
+            return self._data.domain.labels
+        return self._default_labels
+
+    @property
+    def origin(self):
+        return self._origin_before_translation
+
+    @property
+    def scales(self):
+        try:
+            return [d.multiplier for d in self._data.dimension_units]
```

Comment:
The self._data.dimension_units property itself could raise an error, in which case we should just use the default scales.  But also each unit can be `None` to indicate unknown.  In that case we should convert it to `1` as a default.  Similarly for `units`.